### ISSUE TITLE
Disable routing web-traffic as an option

### ISSUE THREAD

**Issue by user0007** [Nyr/openvpn-install#315]
Right now there is no way to disable routing web-traffic through OpenVPN. Many people/developers (IoT) are using OpenVPN as encrypted tunnel to some services but not the web. It will be nice to get this option :-)

Right now, the solution for that use case is:

 - add `route-noexec` to the client configuration file
 - disable `push "dhcp-option DNS X.X.X.X"` in the server configuration file

**Comment by user0007** [comment]
Ok, now I can't connect from server to client :

    server@server:~$ ping 10.8.0.2
        PING 10.8.0.2 (10.8.0.2) 56(84) bytes of data.

and it hangs out. 

So, any idea how to disable routing web-traffic on the server side and see clients from server?

**Comment by Nyr** [comment]
> So, any idea how to disable routing web-traffic on the server side and see clients from server?

Go to the OpenVPN forums for technical support. This is an issue tracker for matters related to the script.

Simplifying things a little bit, you shouldn't use `route-noexec` for that, remove the `push "redirect-gateway def1 bypass-dhcp"` line from the server configuration instead.

But this will not make an ideal setup for your purposes, other things would need to be changed. A road-warrior install has very different requirements than an internal networking one, that's also why this script will not add support for what you want.

**Comment by simonemazzoni** [comment]
@Nyr why this has been closed?
I'm interested too in this "feature". Up to now I have to manually edit the server config to disable web-traffic routing, since I don't need it, almost nullifying the purpose of the script.

> other things would need to be changed

What are those things? If you provide guidelines I can help in the editing of the script.

**Comment by Nyr** [comment]
> why this has been closed?

Technical support was requested and provided to the user and user was told to go through the proper channels to get further support if needed.

The message included also a feature request which is fine to place in the issue tracker, but as explained there are no plans for implementing that specific request.

> closed PR

The PR was closed mostly because of its "works for me" nature. It was very incomplete and not production ready, additional to other reasons which are not very important given the first.

> What are those [other] things [which would need to be changed]?

Incomplete list of what I can think of right now:

- IP forwarding wouldn't be needed so that part needs to be removed, both for the installation and the removal procedures.
- NAT doesn't need to be set up either.
- iptables and firewalld rules would need to be adapted to the new situation (for example, explicitly allow intra-VPN traffic)
- Other configuration options specific for that kind of setup need to be implemented (for example, `client-to-client` is likely desirable).

TL;DR is that I don't think this option will be implemented, at least not in the near future. If I consider that a suggested feature is worth including, I don't have a problem writing the code for it.

**Comment by simonemazzoni** [comment]
@Nyr ok, I kinda see your point. The PR is a fast yet not clean workaround, but I guess has been done due to your refusal to address this specific "configuration".

I think I will fork the project and have a look myself at it, taking into account the list you provided, because I really like this script but I also really don't need intra traffic for my purposes.

**Comment by Nyr** [comment]
My refusal is because a kinda big part of the script needs to be reworked to properly satisfy the needs of a small part of the user base. I get your point and I get that some others are interested too, but I don't feel motivated enough to approach that at this time.

A fork of the script with a proper internal networking configuration and maybe even some specific options for that use case seems like a good idea to me, if someone wants to take the time to do it properly.

For those who don't want that effort, removing that single line can be a quick fix, but not something which I want to make an official "feature" ;)
